### ISSUE TITLE
HLSL: Overload resolution issue

### ISSUE THREAD

**Issue by TiemoJung** [KhronosGroup/glslang#655]
Hi,

we encountered an overload resolution issue:

Example:
```
Texture2D Texture;
SamplerState Sampler;
float3 Coord;
float Level;
float value = Texture.SampleLevel(Sampler, Coord, Level).x;
```

Error:

> ERROR: 'SampleLevel' : ambiguous best function under implicit type conversion 
> ERROR: 'swizzle or member' : Expected 
> ERROR: 'initializer' : Expected 

As it seems the call gets internally turned into:
SampleLevel(Texture, Sampler, Coord, Level)

This is then a problem, because the overload resolution can either (wrongly) pick
SamplerLevel(Texture3D, float3, float) or correctly
SamplerLevel(Texture2D, float2, float) and do the truncation for the Coord parameter.

But as Texture is a Object and SampleLevel is a method, the first pick can not be in the overload set.
So there needs to be an option to fixate the parameter type of methods to avoid this.

**Comment by TiemoJung** [comment]
Hi,

pull request #656 contains a approach to handle the observed misbehavior better.

**Comment by johnkslang** [comment]
This currently works correctly, likely fixed by #656, but wasn't closed when #656 was merged.
